### PR TITLE
test/others: Refactor loop process

### PR DESCRIPTION
**Pull request by rst0git**
This pull request is a follow-up from the discussion in https://github.com/checkpoint-restore/criu/pull/1491. It aims to replace the `loop.sh` script with a `loop` program that is better suited for the tests.

### CODE REVIEW

File: test/others/loop.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	pid_t sid;
+
+	pid = fork();
+	if (pid < 0) {
+		perror("fork failed!");
+		return -1;
+	}
+
+	if (pid == 0) {
+		sid = setsid();
```
Comment (avagin) [inline test/others/loop.c:19]: but the parent process doesn't wait when a child will do setsid() and other actions. Did I miss something?

File: test/others/loop.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#include <stdio.h>
+
+int main() {
+	pid_t pid;
+	pid_t sid;
+
+	pid = fork();
+	if (pid < 0) {
+		perror("fork failed!");
+		return -1;
```
Comment (avagin) [inline test/others/loop.c:15]: return 1;

Comment (rst0git) [reply test/others/loop.c]: Fixed

File: test/others/loop.c (hunk 1 above)
Comment (rst0git) [reply test/others/loop.c:28]: I've updated the commit to wait for the child process to complete the setup.

File: test/others/loop.c [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	}
+
+	if (pid == 0) {
+		close(start_pipe[0]);
+
+		sid = setsid();
+		if(sid < 0) {
+			perror("setsid failed!");
+			res = EXIT_FAILURE;
+			write(start_pipe[1], &res, 1);
```
Comment (avagin) [inline test/others/loop.c:32]: write(..., sizeof(res))

File: test/others/rpc/Makefile [hunk 4]
```diff
@@ -7,6 +7,7 @@ LDLIBS +=  -lprotobuf-c
 PYTHON ?= python
 
 run: all
+	@make -C ..
```
Comment (avagin) [inline test/others/rpc/Makefile:10]: nit: make -C loop

File: test/others/loop.c (hunk 3 above)
Comment (rst0git) [reply test/others/loop.c]: Fixed

File: test/others/rpc/Makefile (hunk 4 above)
Comment (rst0git) [reply test/others/rpc/Makefile]: Updated

**Pull request merged**